Tests/scripts/lock_cloud_machines.py

```diff
@@ -0,0 +1,359 @@
+    Args:
+        storage_bucket(google.cloud.storage.bucket.Bucket): google storage bucket where lock machine is stored.
+        lock_machine_name(str): the name of the file with the list of the test machines.
+        test_machines_path(str): all the exiting machines.
+        gcs_locks_path(str): the lock repository name.
+        number_machines_to_lock(int): the number of the requested machines.
+
+    Returns: the machines list.
+    """
+    if lock_machine_name:  # For debugging: We got a name of a specific machine to use.
+        logging.info('trying to lock the given machine: {options.lock_machine_name}')
+        list_machines = [lock_machine_name]
+    else:
+        logging.info('getting all machine names')  # We are looking for a free machine in all the available machines.
+        test_machines_list = storage_bucket.blob(f'{gcs_locks_path}/{test_machines_path}')
+        list_machines = test_machines_list.download_as_string().decode("utf-8").split()
+        random.shuffle(list_machines)
+
+    if number_machines_to_lock > len(list_machines):
+        logging.error(
+            f'This build requested {number_machines_to_lock} but there are only {len(list_machines)} of active machines')
+        raise
+    return list_machines
+
+
+def wait_for_build_to_be_first_in_queue(storage_client, storage_bucket, gcs_locks_path, job_id, gitlab_status_token):
+    """
+    Run until the job-id is the first in the queue.
+    Args:
+        storage_client(storage.Client): The GCP storage client.
+        storage_bucket(google.cloud.storage.bucket.Bucket): google storage bucket where lock machine is stored.
+        gcs_locks_path(str): the lock repository name.
+        job_id(str): the job id to check.
+        gitlab_status_token(str): the gitlab token.
+    """
+    first_in_the_queue = False
+    while not first_in_the_queue:
+        my_place_in_the_queue, previous_build = get_my_place_in_the_queue(storage_client, gcs_locks_path, job_id)
+        logging.info(f'My place in the queue is: {my_place_in_the_queue}')
+
+        if my_place_in_the_queue == 0:
+            first_in_the_queue = True
+        else:
+            # we check the status of the build that is ahead of me in the queue
+            previous_build_status = check_job_status(gitlab_status_token, previous_build)
+            if previous_build_status != 'running':
+                # delete the lock file of the build because its not running
+                remove_build_from_queue(storage_bucket, gcs_locks_path, previous_build)
+            else:
+                sleep(random.randint(8, 13))
+    return first_in_the_queue
+
+
+def main():
+    install_logging('lock_cloud_machines.log', logger=logging)
+    logging.info('Starting to search for a CLOUD machine/s to lock')
+    options = options_handler()
+    storage_client = storage.Client.from_service_account_json(options.service_account)
+    storage_bucket = storage_client.bucket('xsoar-ci-artifacts')
```

Comment:
Please use as constant:
LOCKS_BUCKET = 'xsoar-ci-artifacts'

```suggestion
    storage_bucket = storage_client.bucket(LOCKS_BUCKET)
```